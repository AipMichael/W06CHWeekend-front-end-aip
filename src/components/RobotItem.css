.card-body {
  background-color: #ffce00;
}

.robot-picture {
  object-fit: cover;
  width: 100%;
}

button {
  margin: 1px;
}
