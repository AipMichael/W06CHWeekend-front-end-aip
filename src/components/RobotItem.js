const RobotItem = ({ robot }) => {
  return (
    <li>
      <div className="card">
        <img className="card-img-top" src={robot.imageUrl} alt="cute robot" />
        <div className="card-body">
          <h5 className="card-title">{robot.name}</h5>
          <p className="card-text">
            Speed:{robot.specifications.speed}
            Toughness:{robot.specifications.toughness}
          </p>
          <button className="btn btn-primary">Edit</button>
          <button href="#" className="btn btn-primary">
            Delete
          </button>
        </div>
      </div>
    </li>
  );
};

export default RobotItem;
