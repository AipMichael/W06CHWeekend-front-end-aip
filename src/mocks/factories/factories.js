import { datatype } from "faker";
import { lorem } from "faker/locale/es";
import { Factory } from "fishery";

const factory = Factory.define(({ sequence }) => ({
  _id: sequence,
  name: lorem.sentence(2),
  imageUrl: datatype.string(1),
  specifications: {
    speed: datatype.number(0, 10),
    toughness: datatype.number(0, 10),
    creationDate: datatype.datetime,
  },
}));

export const getRandomRobot = () => factory.build();
export const getRandomRobots = (count = 3) => factory.buildList(count);
