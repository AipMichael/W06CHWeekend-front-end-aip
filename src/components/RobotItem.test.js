import { render, screen } from "@testing-library/react";
import ReactTestRenderer from "react-test-renderer";
import RobotItem from "./RobotItem";
import { getRandomRobot } from "../mocks/factories/factories";

describe("Given a RobotItem component", () => {
  describe("When it receives a robot", () => {
    test("Then it should render a card with the given robot inside", () => {
      const myRobot = getRandomRobot();

      const cardComponent = ReactTestRenderer.create(
        <RobotItem robot={myRobot} key={myRobot._id} />
      );

      expect(cardComponent.toJSON()).toMatchSnapshot();
    });
  });
});
