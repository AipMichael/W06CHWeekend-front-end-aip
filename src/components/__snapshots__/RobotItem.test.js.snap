// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RobotItem component When it receives a robot Then it should render a card with the given robot inside 1`] = `
<li
  className="col-4"
>
  <div
    className="card"
  >
    <img
      alt="cute robot"
      className="card-img-top robot-picture"
      height="200"
      src="https://i.pinimg.com/236x/04/7a/cd/047acd102d3424cd44743e803818f614--halloween-costumes-th-doctor.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Clockwork Droid
      </h5>
      <p
        className="card-text"
      >
        Speed:
        8
        Toughness:
        6
      </p>
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</li>
`;
